### PR TITLE
[Enhancement] Log on process exit caused by signal

### PR DESCRIPTION
**Pull request by rickif**
## Problem Summary:
Fixes # (issue)
When the BE gets the signal SIGTERM/SIGINT, the process will exit soon without any log.
It's hard to distinguish it from an abnormal exit.
This PR adds log to indicate exit lead by the signal handler.
```
I0517 22:19:45.260324 723345 daemon.cpp:249] got signal: Terminated from pid: 994050320, is going to exit
```
```
I0517 22:21:36.337054 733289 daemon.cpp:249] got signal: Interrupt from pid: 1105230360, is going to exit
```
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/common/daemon.cpp [hunk 1]
```diff
@@ -245,14 +245,15 @@ static void init_starrocks_metrics(const std::vector<StorePath>& store_paths) {
     StarRocksMetrics::instance()->initialize(paths, init_system_metrics, disk_devices, network_interfaces);
 }
 
-void sigterm_handler(int signo) {
+void sigterm_handler(int signo, siginfo_t* info, void* context) {
+    LOG(INFO) << "got signal: " << strsignal(signo) << " from pid: " << info->si_pid << ", is going to exit";
```
Comment (trueeyu) [inline be/src/common/daemon.cpp:249]: LOG(ERROR)?

Comment (rickif) [reply be/src/common/daemon.cpp]: Updated.

**Review by trueeyu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by rickif** [comment]
@mergifyio backport branch-3.0 branch-2.5

**Comment by rickif** [comment]
@Mergifyio backport branch-3.0

**Comment by rickif** [comment]
@Mergifyio backport branch-3.0

**Comment by letian-jiang** [comment]
Why google signal handler is unable to handle this? I thought we have installed this.

```c++
// Install a signal handler that will dump signal information and a stack
// trace when the program crashes on certain signals.  We'll install the
// signal handler for the following signals.
//
// SIGSEGV, SIGILL, SIGFPE, SIGABRT, SIGBUS, and SIGTERM.
GOOGLE_GLOG_DLL_DECL void InstallFailureSignalHandler();
```

**Pull request merged**